include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -63,12 +64,16 @@ namespace RAJA {
 
 #define RAJA_CUDA_REDUCE_BLOCK_LENGTH (1024 + 8) * 16
 
+
 ///
 /// Typedef defining common data type for RAJA-Cuda reduction data blocks
 /// (use this in all cases to avoid type confusion).
 ///
 typedef double CudaReductionBlockDataType;
 
+typedef struct{double val; Index_type idx;} CudaReductionLocBlockDataType; 
```

Comment:
The actual block type is defined in the CPU_MemUtils.hxx file.  We should be able to to use the template type of the Reduction class for this here.  The only caveat would be that users would only be guaranteed correct behavior if their choice of template type has a smaller size than the block type.